unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -1869,6 +1835,83 @@ handleDiffNamespaceToPatch description input = do
     description
     (Path.unabsolute patchPath, Branch.modifyPatches patchName (const patch))
 
+handleIOTest :: HQ.HashQualified Name -> Cli ()
+handleIOTest main = do
+  Cli.Env {codebase, runtime} <- ask
+
+  let testType = Runtime.ioTestType runtime
+  parseNames <- (`NamesWithHistory.NamesWithHistory` mempty) <$> basicParseNames
+  -- use suffixed names for resolving the argument to display
+  ppe <- suffixifiedPPE parseNames
+  let oks results =
+        [ (r, msg)
+          | (r, Term.List' ts) <- results,
+            Term.App' (Term.Constructor' (ConstructorReference ref cid)) (Term.Text' msg) <- toList ts,
+            cid == DD.okConstructorId && ref == DD.testResultRef
+        ]
+      fails results =
+        [ (r, msg)
+          | (r, Term.List' ts) <- results,
+            Term.App' (Term.Constructor' (ConstructorReference ref cid)) (Term.Text' msg) <- toList ts,
+            cid == DD.failConstructorId && ref == DD.testResultRef
+        ]
+
+  matches <-
+    Cli.label \returnMatches -> do
+      -- First, look at the terms in the latest typechecked file for a name-match.
+      whenJustM Cli.getLatestTypecheckedFile \typecheckedFile -> do
+        let refToType :: Reference -> Type Symbol Ann
+            refToType ref =
+              typecheckedFile
+                & UF.indexByReference
```

Comment:
I think this might be doing more work than necessary here; at a glance it looks like we can use `UF.hashTermsId` directly.

Reply:
Yeah, I think that would be better. I was trying to leverage existing stuff (converting a unison file to a `Names`, inverting the `hashTermsId` to lookup by reference instead of name), but I agree it's meh as-written.